docs/ref/forms/api.txt

```diff
@@ -1210,24 +1210,39 @@ Methods of ``BoundField``
 
 .. method:: BoundField.label_tag(contents=None, attrs=None, label_suffix=None)
 
+    Renders a label tag of a form field using the template specified by
+    :attr:`.Form.template_name_label`.
+
+    The available context is:
+
+    * ``field``: This instance of the :class:`BoundField`.
+    * ``contents``: By default a concatenated string of
+      :attr:`BoundField.label` and :attr:`Form.label_suffix`. This can be
+      overridden by the ``contents`` and ``label_suffix`` arguments.
+    * ``attrs``: A ``dict`` containing ``for``, :attr:`Form.required_css_class`
+      and ``id``. ``id`` is generated by the field's widget ``attrs`` or
+      :attr:`Form.auto_id`. Additional attributes can be provided by the
+      ``attrs`` argument.
+    * ``use_tag``: A boolean which is ``True`` if the label has an ``id``.
+      If ``False`` the default template omits the ``<label>`` tag.
+
+    .. tip::
+
+        In your template ``field`` is the instance of the ``BoundField``.
+        Therefore ``field.field`` accesses :attr:`BoundField.field` being
+        the field you declare, your ``forms.CharField`` for example.
+
     To separately render the label tag of a form field, you can call its
     ``label_tag()`` method::
 
         >>> f = ContactForm(data={'message': ''})
         >>> print(f['message'].label_tag())
         <label for="id_message">Message:</label>
 
-    You can provide the ``contents`` parameter which will replace the
-    auto-generated label tag. An ``attrs`` dictionary may contain additional
-    attributes for the ``<label>`` tag.
-
-    The HTML that's generated includes the form's
-    :attr:`~django.forms.Form.label_suffix` (a colon, by default) or, if set,
-    the current field's :attr:`~django.forms.Field.label_suffix`. The optional
-    ``label_suffix`` parameter allows you to override any previously set
-    suffix. For example, you can use an empty string to hide the label on
-    selected fields. The label is rendered using the template specified by the
-    forms :attr:`.Form.template_name_label`.
+    If you'd like to customize the rendering this can be achieved by overriding
+    the :attr:`.Form.template_name_label` attribute or more generally by
+    overriding the default template, see also
+    :ref:`overriding-built-in-form-templates`.
```

Comment:
This may be fine as-is for the audience, but just to raise it anyway: overriding the `template_name_label` alone _won't_ achieve anything, you've got to do the rest of the [currently implied] work, which is `overriding the attribute ... to point to a custom template file`
(Unless you just do `template_name_label = Template(...)` which I presume would work, but no-one is likely to do)
Having now spelt it out though, I expect if you're at the point where you're customising such things, you can already infer what it means to change the attribute, so it probably _is_ fine as-is.

Reply:
I think it's fine to keep the current form.